Framework update

DeborahVolpe opened the pull request:
Framework update for the quantum week presentation

burgholzer reviewed (approved):
Just did a quick round of fixes and updated some things. See the individual commits for details.
Overall, this should be good to go.
You might want to also add your new QCE publication to the `docs/refs.bib` and cite it in the documentation.

One thing i noticed is that the `solvers.py` file contains huge amounts of code duplication and is probably 5x longer than it needs to be. However, this is nothing to be changed in this PR, but maybe something to keep in mind for a potential future refactor.

DeborahVolpe commented:
> Just did a quick round of fixes and updated some things. See the individual commits for details. Overall, this should be good to go. You might want to also add your new QCE publication to the `docs/refs.bib` and cite it in the documentation.
> Thank you so much. I have also added the reference here. I have forgotten this point 

> One thing i noticed is that the `solvers.py` file contains huge amounts of code duplication and is probably 5x longer than it needs to be. However, this is nothing to be changed in this PR, but maybe something to keep in mind for a potential future refactor.

Yes, thank you.
